plugin/src/main/java/io/jenkins/plugins/casc/SchemaGeneration.java

```diff
@@ -199,15 +199,26 @@ private static void generateEnumAttributeSchema(JSONObject attributeSchemaTempla
     }
 
     public static void rootConfigGeneration() throws Exception {
+
+
         DefaultConfiguratorRegistry registry = new DefaultConfiguratorRegistry();
         final ConfigurationContext context = new ConfigurationContext(registry);
+        context.setMode("JSONSchema");
         for (RootElementConfigurator root : RootElementConfigurator.all()) {
             final CNode config = root.describeStructure(root.getTargetComponent(context), context);
             final Mapping mapping = config.asMapping();
             final List<Map.Entry<String, CNode>> entries = new ArrayList<>(mapping.entrySet());
```

Comment:
@casz `globalNodeProperties NodeProperty` The mapping is as follows, so we are not given back the entire structure as such, whereas what we need is an iterable NodeProperty which at the moment is not possible.
`  

      globalNodeProperties:

    - envVars:

        env:

          - key: FOO

            value: BAR
`

Reply:
can you add a failing test or  point to one  I should run please?

Reply:
is this what you're after?
```
                CNode value = entry.getValue();
                if (value.getType() == Type.SEQUENCE) {
                    Sequence cNodes = value.asSequence();
                    cNodes.forEach(node -> { 
                        
                    });
                }
```

Reply:
Not really because we wont enter this if statement at all.

Reply:
is it not a sequence?

Reply:
No it isnt.

Reply:
Moreover it is not iteratable  because we cannot map it as a sequence

Reply:
@timja any ideas on this??

Reply:
I'll need some type to take a look at it

Reply:
Isn't this the issue? 
https://github.com/jenkinsci/configuration-as-code-plugin/blob/81b9a6857b03307d77bb58bdfa2e348c8ee4eed3/plugin/src/main/java/io/jenkins/plugins/casc/Configurator.java#L181

You aren't describing  the structure at all you're just adding a scalar value, that should be a sequence of NodeProperty here not a scalar?

Reply:
I think if `attribute.isMultiple()`  returns true on that line you should be using a sequence?

Reply:
I guess the mapping on the original describe https://github.com/jenkinsci/configuration-as-code-plugin/blob/81b9a6857b03307d77bb58bdfa2e348c8ee4eed3/plugin/src/main/java/io/jenkins/plugins/casc/Configurator.java#L165 includes a value so we want to mimic that right just with the type instead of the value?

Reply:
You’re going to need to indicate that it’s a sequence some how, where I pointed is where the information is lost

Reply:
You’re going to need to indicate that it’s a sequence some how, where I pointed is where the information is lost

Reply:
Do you mean that we should be iterating over the attribute type and keep adding it to the mapping something like this
if (attribute.isMultiple == true) {
 Iterate over the sequence and add it to the mapping
}

Reply:
If we indicate the mapping as a sequence would it make it easier to iterate further?